### PR TITLE
WP-1300: Normalized errors and removed unnecessary required checks

### PR DESCRIPTION
**Pull request by ErikIDX**
Normalized errors from IDX's API and stripped HTML from the error codes. Also removes unneeded `required` checks for non-required args.

### CODE REVIEW

File: idx/admin/rest-controller.php [hunk 1]
```diff
@@ -108,16 +108,39 @@ public function authorization_failed_status_code() {
...
+			return $error;
+		}
+		$code    = $error->get_error_code();
+		$message = $error->get_error_message();
+		$data    = $error->get_error_data();
+		if ( 'idx_api_error' === $code ) {
+			return new \WP_Error(
+				$code,
+				$data['rest_error'],
+				array(
```
Comment (ralls) [inline idx/admin/rest-controller.php:137]: PHP 5.4 introduced shorthand arrays - any reason to not use them?

Comment (ErikIDX) [reply idx/admin/rest-controller.php:137]: Technically against WPCS, but after talking with Drew we decided to write exceptions into the linter and use the shorthand -- will update across the board for the recent changes in a PR after this one.

**Review by ralls — APPROVED** [review]
👍

File: idx/admin/rest-controller.php (hunk 1 above)
Comment (ralls) [reply idx/admin/rest-controller.php:137]: Seems like an archaic standard now that even php 7.x is EOL 🤷‍♂️

**Pull request merged**